esinstall/src/util.ts

```diff
@@ -190,8 +190,23 @@ export function createInstallTarget(specifier: string, all = true): InstallTarge
   };
 }
 
-/** Is this file JavaScript?  */
-export function isJavaScript(pathname: string) {
+/**
+ * Detect the web dependency "type" as either JS or ASSET:
+ *   - BUNDLE: Install and bundle this file with Rollup engine.
+ *   - ASSET: Copy this file directly.
+ */
+export function getWebDependencyType(pathname: string): 'ASSET' | 'BUNDLE' {
   const ext = path.extname(pathname).toLowerCase();
-  return ext === '.js' || ext === '.mjs' || ext === '.cjs';
+  // JavaScript should always be bundled.
+  if (ext === '.js' || ext === '.mjs' || ext === '.cjs') {
+    return 'BUNDLE';
+  }
+  // Svelte & Vue should always be bundled because we want to show the missing plugin
+  // error if a Svelte or Vue file is the install target. Without this, the .svelte/.vue
+  // file would be treated like an asset and sent to the web as-is.
+  if (ext === '.svelte' || ext === '.vue') {
+    return 'BUNDLE';
+  }
+  // All other files should be treated as assets (copied over directly).
+  return 'ASSET';
```

Comment:
So do we not need to handle JSX/TSX because of built-in support from esbuild? Curious if that’s really a different strategy than this (though JSX not carrying CSS seems different enough on its own)

Reply:
> I don't think we have a real use-case for a user publishing JSX/TSX in a package.

I can say myself that I’ve tried, and been punished by webpack 😅 . I think just as a consequence of the ecosystem I’d guess most popular React packages have been forced to prebundle JSX and this statement is correct

Reply:
yea that's fair enough, and we want that error to be reported instead of the JSX file copied over 1:1 as if it were an image.

+1 for a PR to add JS, JSX, TS, and TSX here to this function AND add a curated `JSX found, add this plugin to support` error message like we have for Vue and Svelte.